conformance/reports/v1.0.0/azure-application-gateway-for-containers/v1.0.0-report.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: gateway.networking.k8s.io/v1alpha1
+date: "2024-04-24T16:38:02-07:00"
+gatewayAPIVersion: v1.0.0
+implementation:
+  contact:
+  - agcfeedback@microsoft.com
+  organization: Microsoft Azure
+  project: Application Gateway for Containers
+  url: https://aka.ms/appgwcontainers/docs
+  version: v1.0.0
+kind: ConformanceReport
+profiles:
+- core:
+    result: partial
+    statistics:
+      Passed: 28
+      Skipped: 1
+      Failed: 0
+    summary: ""
+    skippedTests:
+    - GatewayClassObservedGenerationBump
```

Comment:
We don't really have a lot of prior art for how we represent the idea of conformance status when it comes to skipped tests. Apparently we actually have another implementation that has a skipped test in their report, but I think we just missed that:

https://github.com/kubernetes-sigs/gateway-api/blob/7320257302905254dcebf4f5c424741cf1556c06/conformance/reports/v1.0.0/istio-istio/1.20-report.yaml#L31

This doesn't need to block this PR, but I'll create a follow up issue to track how we can represent this kind of partial conformance on our implementations page.

/cc @mlavacca @shaneutt

Reply:
We already have rules in our rulebook about partial results

https://github.com/kubernetes-sigs/gateway-api/blob/7320257302905254dcebf4f5c424741cf1556c06/conformance/reports/README.md#L123-L127

The rule here should be applied starting from 1.1.0, but I honestly think it makes sense to begin enforcing it here from now on, even considering the skipped test is a core one.